R/uniDimBayesianCalcCoeff.R

```diff
@@ -569,3 +543,107 @@
   callback()
   return(out)
 }
+
+
+.BayesianComputeScaleResults <- function(jaspResults, options, model) {
+  if (!is.null(.getStateContainerB(jaspResults)[["scaleResultsObj"]]$object))
+    return(.getStateContainerB(jaspResults)[["scaleResultsObj"]]$object)
+
+  out <- model[["scaleResults"]]
+  if (is.null(out))
+    out <- list()
+
+  if (is.null(model[["empty"]])) {
+
+    ciValue <- options[["credibleIntervalValueScale"]]
+
+    selected <- names(which(model[["derivedOptions"]][["selectedEstimators"]]))
+    # first the coefficients with samples
+    samps <- model[selected]
+    samps <- lapply(samps, function(x) x[["samp"]])
+    samps[sapply(samps, is.null)] <- NULL
+
+    out[["est"]] <- lapply(samps, mean)
+    out[["cred"]] <- lapply(samps, function(x) coda::HPDinterval(coda::mcmc(c(x)), prob = ciValue))
+
+    if (options[["rHat"]]) {
+      tmp <- lapply(samps, function(x) {lapply(as.data.frame(t(x)), coda::mcmc)})
+      out[["rHat"]] <- lapply(tmp, function(x) {coda::gelman.diag(coda::as.mcmc.list(x))[["psrf"]][, 1]})
```

Comment:
how about:
```suggestion
      lapply(samps, function(x) {
        # for each samp, (1) convert the rows to a coda object, (2) convert to mcmc.list, so that (3) gelman.diag is happy.
        coda::gelman.diag(coda::as.mcmc.list(lapply(seq_len(nrow(x)), function(i) coda::mcmc(x[i, ]))))[["psrf"]][, 1]
      })
```
with nested `lapply`s, a comment can be very useful for future readers. Avoiding `as.data.frame(t(x))` should also be slightly more efficient.